### PR TITLE
feat: Sequencer gets configurable iteration callback

### PR DESCRIPTION
**Pull request by paulgessinger**
The default is a noop function pointer.
The background is that for the python bindings, I need to hook into the
event loop to be able to correctly respond to Ctrl+C. This is the most
decoupled way to do this I could think of.

Also increases the amount of output `Sequencer` produces in verbose
mode.

### CODE REVIEW

File: Examples/Framework/src/Framework/Sequencer.cpp [hunk 1]
```diff
@@ -290,21 +291,25 @@ int ActsExamples::Sequencer::run() {
...
               if (rdr->read(++context) != ProcessCode::SUCCESS) {
                 throw std::runtime_error("Failed to read input data");
               }
             }
-            // Execute all algorithms
+
+            ACTS_VERBOSE("Execute algorithms");
             for (auto& alg : m_algorithms) {
               StopWatch sw(localClocksAlgorithms[ialgo++]);
+              ACTS_VERBOSE("Execute Algorithm: " << alg->name());
```
Comment (HadrienG2) [inline Examples/Framework/src/Framework/Sequencer.cpp:306]: It feels somewhat strange that algorithm names get spelled out, but not reader/writer names. May want to add those as well?

**Review by HadrienG2 — CHANGES_REQUESTED** [review]

File: Examples/Framework/include/ActsExamples/Framework/Sequencer.hpp [hunk 2]
```diff
@@ -26,6 +26,8 @@
 
 namespace ActsExamples {
 
+using IterationCallback = void (*)();
```
[the remark this answers is not in the document]
Comment (paulgessinger) [reply Examples/Framework/include/ActsExamples/Framework/Sequencer.hpp:29]: Right. The thing is I don't need any additional functionality and I wasn't convinced I should implement a more featured interface if there are zero users for it. I could add the event number for good measure but that's unused for the bindings. 

What do you think?

File: Examples/Framework/include/ActsExamples/Framework/Sequencer.hpp [hunk 3]
```diff
@@ -44,6 +46,7 @@ class Sequencer {
     int numThreads = -1;
     /// output directory for timing information, empty for working directory
     std::string outputDir;
+    IterationCallback iterationCallback = []() {};
```
[the remark this answers is not in the document]
Comment (paulgessinger) [reply Examples/Framework/include/ActsExamples/Framework/Sequencer.hpp:49]: True, good point, will do.

File: Examples/Framework/src/Framework/Sequencer.cpp (hunk 1 above)
Comment (paulgessinger) [reply Examples/Framework/src/Framework/Sequencer.cpp:306]: I can add that no problem.

File: Examples/Framework/include/ActsExamples/Framework/Sequencer.hpp (hunk 2 above)
[the remark this answers is not in the document]
Comment (HadrienG2) [reply Examples/Framework/include/ActsExamples/Framework/Sequencer.hpp:29]: I'm fine with keeping this as-is and adding functionality later as needed.

[the remark this answers is not in the document]
Comment (paulgessinger) [reply Examples/Framework/include/ActsExamples/Framework/Sequencer.hpp:29]: Ok

File: Examples/Framework/include/ActsExamples/Framework/Sequencer.hpp (hunk 3 above)
[the remark this answers is not in the document]
Comment (paulgessinger) [reply Examples/Framework/include/ActsExamples/Framework/Sequencer.hpp:51]: Added.

File: Examples/Framework/src/Framework/Sequencer.cpp (hunk 1 above)
Comment (paulgessinger) [reply Examples/Framework/src/Framework/Sequencer.cpp]: Added verbose output for context decorators, readers and writers.

**Review by HadrienG2 — APPROVED** [review]
Looks good to me!

**Pull request merged**